agent/inbound.go

```diff
@@ -161,8 +160,7 @@ func (a *Agent) deleteApplication(app *v1alpha1.Application) error {
 
 	logCtx.Infof("Deleting application")
 
-	delPol := v1.DeletePropagationBackground
-	err := a.appclient.ArgoprojV1alpha1().Applications(a.namespace).Delete(a.context, app.Name, v1.DeleteOptions{PropagationPolicy: &delPol})
+	err := a.appManager.Delete(a.context, a.namespace, app, true)
```

Comment:
I verified that the likely reason we were using appclient Delete call here (rather than appManager.Delete) is for DeletePropagationBackground. I've thus added it to app manager.